File: 3621-minimum-operations-to-make-array-values-equal-to-k/3621-minimum-operations-to-make-array-values-equal-to-k.cpp
class Solution {
   public:
    int minOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        if (nums[0] < k) return -1;

        nums.erase(unique(nums.begin(), nums.end()), nums.end());
        return nums[0] > k ? nums.size() : nums.size() - 1;
    }
};

File: 3621-minimum-operations-to-make-array-values-equal-to-k/3621-minimum-operations-to-make-array-values-equal-to-k.go
func minOperations(nums []int, k int) int {
    mp := make(map[int]bool)    
    minima := int(1e9)
    for _, num := range nums {
        mp[num] = true
        minima = min(minima, num)
    }
    if minima < k {
        return -1
    } 

    if minima > k {
        return len(mp)
    }
    return len(mp)-1
}